### ISSUE TITLE
Occasionally, app crash caused by SDK

### ISSUE THREAD

**Issue by ltluc253** [dji-sdk/Mobile-SDK-Android#164]
#### Is this issue about bugs or crash issues of DJI Android SDK and Sample Code?

No, It crash on my app. But I got stacktrace from DJI SDK as below:

> Fatal Exception: net.sqlcipher.database.SQLiteException: no such table: polygon_tfr_remote_data: DELETE FROM polygon_tfr_remote_data
>        at net.sqlcipher.database.SQLiteDatabase.native_execSQL(Unknown Source)
>        at net.sqlcipher.database.SQLiteDatabase.execSQL(Unknown Source)
>        at dji.thirdparty.afinal.FinalDb.deleteAll(Unknown Source)
>        at dji.midware.data.forbid.db.FlyfrbBaseDbManager.deleteAll(Unknown Source)
>        at dji.internal.geofeature.flyforbid.FlyfrbPolygonDataUpdateManager$1.onSuccess(Unknown Source)
>        at dji.internal.geofeature.flyforbid.FlyfrbPolygonDataUpdateManager$1.onSuccess(Unknown Source)
>        at dji.thirdparty.afinal.http.HttpHandler.onProgressUpdate(Unknown Source)
>        at dji.thirdparty.afinal.core.AsyncTask$InternalHandler.handleMessage(Unknown Source)
>        at android.os.Handler.dispatchMessage(Handler.java:102)
>        at android.os.Looper.loop(Looper.java:154)
>        at android.app.ActivityThread.main(ActivityThread.java:6119)
>        at java.lang.reflect.Method.invoke(Method.java)
>        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
>        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)

#### My test environment as below:
 - DJI SDK 4.2.1
 - Tablet Nexus 9
 - Android 7.1.1
 - Phantom 3 Pro, Phantom 4, Phantom 4 Pro

**Comment by zoemcateedji** [comment]
Thanks for the feedback - this is fixed in our upcoming 4.3 release.

**Comment by zoemcateedji** [comment]
Could you please try with the new 4.3 build and let us know how it goes?

https://developer.dji.com/mobile-sdk/downloads/

**Comment by nextfullstorm** [comment]
Got it on 4.3.2 several times.

SDK 4.3.2
Android 6.0
Phantom 4

**Comment by zoemcateedji** [comment]
Interesting! When you say several times, does it repro only after a clean install, or any other time?

**Comment by nextfullstorm** [comment]
Guys, you should really look into your flyforbid code.

> =====	Device Info	=====
> manufacture:LENOVO
> product:TB-8703F
> model:Lenovo TB-8703F
> version:TB-8703F_S000029_170113_ROW
> android version:6.0.1
> sdk version:23
> =====	App Info	=====
> versionCode:1
> versionName:1.0
> =====	Crash	=====
> java.lang.IllegalStateException: database not open
> 	at net.sqlcipher.database.SQLiteDatabase.execSQL(SQLiteDatabase.java:2262)
> 	at dji.thirdparty.afinal.FinalDb.deleteAll(Unknown Source)
> 	at dji.midware.data.forbid.db.FlyfrbBaseDbManager.deleteAll(Unknown Source)
> 	at dji.pilot.flyforbid.c$1.a(Unknown Source)
> 	at dji.pilot.flyforbid.c$1.onSuccess(Unknown Source)
> 	at dji.thirdparty.afinal.http.HttpHandler.onProgressUpdate(Unknown Source)
> 	at dji.thirdparty.afinal.core.AsyncTask$InternalHandler.handleMessage(Unknown Source)
> 	at android.os.Handler.dispatchMessage(Handler.java:102)
> 	at android.os.Looper.loop(Looper.java:148)
> 	at android.app.ActivityThread.main(ActivityThread.java:5417)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by nextfullstorm** [comment]
> Interesting! When you say several times, does it repro only after a clean install, or any other time?

What do you mean by clean install? I make changes in my android app, upload it on tablet and on every 3th - 4th run got 
> net.sqlcipher.database.SQLiteDatabaseCorruptException: database disk image is malformed: DELETE FROM polygon_tfr_remote_data

If I completely delete my app, will the databases reload from your server or smth?

**Comment by JamesMasterman** [comment]
I'm seeing this as well, in SDK 4.3.2

It's fairly random, but I see a couple of different errors which result in a crash. The app was freshly installed at this time, but I don't know if that makes a major difference.

`E/Database: malformed database schema (39.904152): , while compiling: select count(*) from sqlite_master;
                                           net.sqlcipher.database.SQLiteDatabaseCorruptException: malformed database schema (39.904152): , while compiling: select count(*) from sqlite_master;
                                               at net.sqlcipher.database.SQLiteCompiledSql.native_compile(Native Method)
                                               at net.sqlcipher.database.SQLiteCompiledSql.compile(Unknown Source)
                                               at net.sqlcipher.database.SQLiteCompiledSql.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteProgram.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteQuery.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDirectCursorDriver.query(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.rawQueryWithFactory(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.rawQuery(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.keyDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openDatabaseInternal(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openOrCreateDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(Unknown Source)
                                               at dji.thirdparty.afinal.FinalDb.generateDb(Unknown Source)
                                               at dji.thirdparty.afinal.FinalDb.reopen(Unknown Source)
                                               at dji.midware.data.forbid.db.FlyfrbPolygonMainDbManager.replaceDbFile(Unknown Source)
                                               at dji.midware.data.forbid.DJIFlyForbidController.replaceSDKDbFile(Unknown Source)
                                               at dji.sdk.sdkmanager.GeoSyncFileService$1.handleMessage(Unknown Source)
                                               at android.os.Handler.dispatchMessage(Handler.java:98)
                                               at android.os.Looper.loop(Looper.java:154)
                                               at android.os.HandlerThread.run(HandlerThread.java:61)`

also 

`Database: Calling error handler for corrupt database /data/user/0/com...../databases/flysafe_areas_djisdk.db
                                           net.sqlcipher.database.SQLiteDatabaseCorruptException: malformed database schema (39.904152): , while compiling: select count(*) from sqlite_master;
                                               at net.sqlcipher.database.SQLiteCompiledSql.native_compile(Native Method)
                                               at net.sqlcipher.database.SQLiteCompiledSql.compile(Unknown Source)
                                               at net.sqlcipher.database.SQLiteCompiledSql.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteProgram.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteQuery.<init>(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDirectCursorDriver.query(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.rawQueryWithFactory(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.rawQuery(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.keyDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openDatabaseInternal(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteDatabase.openOrCreateDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(Unknown Source)
                                               at net.sqlcipher.database.SQLiteOpenHelper.getWritableDatabase(Unknown Source)
                                               at dji.thirdparty.afinal.FinalDb.generateDb(Unknown Source)
                                               at dji.thirdparty.afinal.FinalDb.reopen(Unknown Source)
                                               at dji.midware.data.forbid.db.FlyfrbPolygonMainDbManager.replaceDbFile(Unknown Source)
                                               at dji.midware.data.forbid.DJIFlyForbidController.replaceSDKDbFile(Unknown Source)
                                               at dji.sdk.sdkmanager.GeoSyncFileService$1.handleMessage(Unknown Source)
                                               at android.os.Handler.dispatchMessage(Handler.java:98)
                                               at android.os.Looper.loop(Looper.java:154)
                                               at android.os.HandlerThread.run(HandlerThread.java:61)
10-15 16:51:19.472 5162-5469/? E/DefaultDatabaseErrorHandler: Corruption reported by sqlite on database, deleting: /data/user/0/com..../databases/flysafe_areas_djisdk.db
10-15 16:51:19.472 5162-5469/? E/DefaultDatabaseErrorHandler: deleting the database file: /data/user/0/com..../databases/flysafe_areas_djisdk.db`

**Comment by JamesMasterman** [comment]
A couple more crashes that I saw just after my app updated via the play store.

This was on a Samsung S7 Edge phone with Android 7

`E/UncaughtException: net.sqlcipher.database.SQLiteDatabaseCorruptException: error code 11: database disk image is malformed
                                                        at net.sqlcipher.database.SQLiteStatement.native_execute(Native Method)
                                                        at net.sqlcipher.database.SQLiteStatement.execute(Unknown Source)
                                                        at net.sqlcipher.database.SQLiteDatabase.execSQL(Unknown Source)
                                                        at dji.thirdparty.afinal.FinalDb.exeSqlInfo(Unknown Source)
                                                        at dji.thirdparty.afinal.FinalDb.save(Unknown Source)
                                                        at dji.midware.data.forbid.db.FlyfrbBaseDbManager.save(Unknown Source)
                                                        at dji.pilot.flyforbid.c$1.a(Unknown Source)
                                                        at dji.pilot.flyforbid.c$1.onSuccess(Unknown Source)
                                                        at dji.thirdparty.afinal.http.HttpHandler.onProgressUpdate(Unknown Source)
                                                        at dji.thirdparty.afinal.core.AsyncTask$InternalHandler.handleMessage(Unknown Source)
                                                        at android.os.Handler.dispatchMessage(Handler.java:102)
                                                        at android.os.Looper.loop(Looper.java:154)
                                                        at android.app.ActivityThread.main(ActivityThread.java:6692)
                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)`

and

`E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: com....., PID: 9829
                                                 net.sqlcipher.database.SQLiteDatabaseCorruptException: error code 11: database disk image is malformed
                                                     at net.sqlcipher.database.SQLiteStatement.native_execute(Native Method)
                                                     at net.sqlcipher.database.SQLiteStatement.execute(Unknown Source)
                                                     at net.sqlcipher.database.SQLiteDatabase.execSQL(Unknown Source)
                                                     at dji.thirdparty.afinal.FinalDb.exeSqlInfo(Unknown Source)
                                                     at dji.thirdparty.afinal.FinalDb.save(Unknown Source)
                                                     at dji.midware.data.forbid.db.FlyfrbBaseDbManager.save(Unknown Source)
                                                     at dji.pilot.flyforbid.c$1.a(Unknown Source)
                                                     at dji.pilot.flyforbid.c$1.onSuccess(Unknown Source)
                                                     at dji.thirdparty.afinal.http.HttpHandler.onProgressUpdate(Unknown Source)
                                                     at dji.thirdparty.afinal.core.AsyncTask$InternalHandler.handleMessage(Unknown Source)
                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                     at android.os.Looper.loop(Looper.java:154)
                                                     at android.app.ActivityThread.main(ActivityThread.java:6692)
                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)`

**Comment by lennoncork** [comment]
We are also seeing this issue.

**Comment by talobin** [comment]
Hi all, 
Thank you for sending the stack trace.
If any of you have a way to consistent replicate this crash please let us know.
Please avoid sending similar stack trace.

**Comment by talobin** [comment]
Will go ahead and close this as it is duplicate of 
https://github.com/dji-sdk/Mobile-SDK-Android/issues/176
